include/c-spiffe/utils/error.h

```diff
@@ -0,0 +1,84 @@
+#ifndef INCLUDE_UTILS_ERROR_H
+#define INCLUDE_UTILS_ERROR_H
+
+#include <errno.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** Error code enum */
+enum enum_err_t {
+    NO_ERROR = 0,
+    ERR_CANCELLED_STATUS,
+    ERR_CLOSING = ERR_CANCELLED_STATUS,
+    ERR_FAILED_OPEN,
```

Comment:
```suggestion
    ERR_OPEN_FILE,
```
Since it's an error already I think there is no need to put "FAILED". If it does not refer only to files, it could be just "ERR_OPEN"